scripts.py

```diff
@@ -872,19 +863,19 @@ def __init__(self, provider, _db=None, cmd_args=None, *provider_args, **provider
 
         super(RunCoverageProviderScript, self).__init__(_db)
         parsed_args = self.parse_command_line(self._db, cmd_args)
-        if callable(provider):
-            if parsed_args.identifier_type:
-                self.identifier_type = parsed_args.identifier_type
-                self.identifier_types = [self.identifier_type]
-            else:
-                self.identifier_type = None
-                self.identifier_types = []
+        if parsed_args.identifier_type:
+            self.identifier_type = parsed_args.identifier_type
+            self.identifier_types = [self.identifier_type]
+        else:
+            self.identifier_type = None
+            self.identifier_types = []
 
-            if parsed_args.identifiers:
-                self.identifiers = parsed_args.identifiers
-            else:
-                self.identifiers = []
+        if parsed_args.identifiers:
+            self.identifiers = parsed_args.identifiers
+        else:
+            self.identifiers = []
 
+        if callable(provider):
```

Comment:
This is totally unrelated -- I found it because I was mistakenly treating the search coverage provider as a CollectionCoverageProvider instead of a WorkCoverageProvider -- but I think it is a real bug fix. Previously, if you passed in an already-instantiated CollectionCoverageProvider object, `identifiers` and `identifier_types` wasn't being set, and you'd get a crash.